src/test/java/wooteco/subway/line/LineDaoTest.java

```diff
@@ -0,0 +1,119 @@
+package wooteco.subway.line;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.List;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.TestConstructor;
+import org.springframework.test.context.jdbc.Sql;
+import wooteco.subway.exception.DuplicateException;
+import wooteco.subway.exception.NotExistItemException;
+
+@SpringBootTest()
+@TestConstructor(autowireMode = TestConstructor.AutowireMode.ALL)
+@Sql("classpath:tableInit.sql")
+@DisplayName("노선 DAO 관련 기능")
+class LineDaoTest {
+
+    private final LineDao lineDao;
+
+    public LineDaoTest(LineDao lineDao) {
+        this.lineDao = lineDao;
+    }
+
+    @Test
+    @DisplayName("노선 한 개가 저장된다.")
+    void save() {
+        Line line = lineDao.save(new Line("2호선", "bg-green-600"));
+
+        assertThat(line.getId()).isEqualTo(1L);
+        assertThat(line.getName()).isEqualTo("2호선");
+        assertThat(line.getColor()).isEqualTo("bg-green-600");
+    }
+
+    @Test
+    @DisplayName("중복된 이름을 갖는 노선은 저장이 안된다.")
+    void duplicateSaveValidate() {
+        Line line = new Line("2호선", "bg-green-600");
+        lineDao.save(line);
+
+        assertThatThrownBy(() -> lineDao.save(line))
+            .isInstanceOf(DuplicateException.class);
+    }
+
+    @Test
+    @DisplayName("모든 노선 목록을 조회한다")
+    void findAll() {
+        Line line2 = lineDao.save(new Line("2호선", "bg-green-600"));
+        Line line3 = lineDao.save(new Line("3호선", "bg-orange-600"));
+        Line line4 = lineDao.save(new Line("4호선", "bg-skyBlue-600"));
+
+        List<Line> lines = lineDao.findAll();
+
+        assertThat(lines).hasSize(3).containsExactly(line2, line3, line4);
```

Comment:
```suggestion
        assertThat(lines).hasSize(3)
            .containsExactly(line2, line3, line4);
```

이렇게 내려주는 편이 읽기 더 좋은 것 같아요.